src/aosm/azext_aosm/_configuration.py

```diff
@@ -107,9 +109,18 @@ def helptext(cls) -> "VhdArtifactConfig":
             image_api_version=(
                 "Optional. The ARM API version used to create the Microsoft.Compute/images resource."
             ),
-            **asdict(ArtifactConfig.helptext()),
+            **asdict(artifact_config),
         )
 
+    def validate(self):
+        """
+        Validate the configuration.
+        """
+        if self.blob_sas_url and self.file_path:
+            raise ValidationError("Only one of file_path or blob_sas_url may be set.")
+        if not (self.blob_sas_url or self.file_path):
+            raise ValidationError("One of file_path or sas_blob_url must be set.")
+
```

Comment:
This should work but I have maintainability concerns about fully overriding the method in this subclass where the base class is also a full class - imagine a scenario in future where new fields are added to `ArtifactConfig` which also apply to `VhdArtifactConfig`; any validation will need to be added to two places and there's risk of it being missed.

It would be better to have VHD artifact config and other artifact config be handled by separate subclasses of a common Abstract Base Class (ABC) where common field validation which always applies to both is handled in the `validate()` method of that class (this would not include `file_path` since its validation differs based on subclass), and each subclass handles its own specific validation after calling the super's version of the method. Look immediately below to the `Configuration` class and its subclasses for a good example of this.

Reply:
I've decided to do the smallest fix now pre-GA and raise another post-GA fix to do this refactor, as felt too close to the deadline (today)